src/test/java/fr/techad/edc/httpd/utils/TokenUtilsTest.java

```diff
@@ -0,0 +1,62 @@
+package fr.techad.edc.httpd.utils;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TokenUtilsTest {
+  private final String tokenPath = "./token.info";
+  private final String keyPath = "./private.key";
+  TokenUtils tutil = TokenUtils.getInstance();
+
+  @Before
+  public void createToken() {
+    tutil.createTokenFile();
+  }
+
+  @After
+  public void removeFakeKey() {
+    FileUtils futil = FileUtils.getInstance();
+    if (futil.readFile(keyPath).equals("IamAfakePrivateK3Y"))
+      futil.writeFile(keyPath, "");
+  }
+
+  @Test
+  public void keyCreationLength() {
+    Assert.assertEquals(24, tutil.genSecretKey().length());
+  }
+
+  @Test
+  public void tokenValidationEmptyTest() {
```

Comment:
This test is not mandatory. Either you manage the null parameter to avoid the exception or you ignore this test because this is the default behavior of the JVM